packages/kit/src/core/adapt/prerender/prerender.js

```diff
@@ -96,12 +96,26 @@ export async function prerender({ cwd, out, log, config, build_data, fallback, a
 		}
 	});
 
+	/**
+	 * Checks if path has extension
+	 * @param {string} path
+	 */
+	function has_ext(path) {
+		// has_ext('.htaccess') === false
```

Comment:
I don't understand. Why do you want a trailing slash on `.htaccess`?

Reply:
We wouldn't want to do that, but I added that in as an example of where a `.file` would have a trailing slash. If you had a route that was `htaccess` (or any name for that matter) it would still add a trailing slash if you used `trailingSlash: 'always'`. This behavior is the same as it was before this PR.

This PR only addresses when the filename in the path "has an extension", and `.htaccess` is not a file with an extension. Thoughts?

Reply:
I think having a per page / per endpoint way to override the value of the `trailingSlash` option would be awesome, and may make this PR unnecessary. I tried looking into how to do that but quickly got overwhelmed. If someone can direct me, I'm happy to take that on.

Reply:
I kind of think dot files should be considered as having an extension by default if we're going to handle files with extensions

If we wanted to make `trailingSlash` more configurable I think it'd have to be a resolve option similar to `ssr`. I'll let @Rich-Harris weigh in on what the best API for this might be though

Reply:
Alternatively, never adding a trailing slash for endpoints makes sense in my opinion, regardless of the `trailingSlash` configuration value. The behavior for pages can be left as is.